### PR TITLE
refactor: visitor list use auto refetch

### PR DESCRIPTION
**Pull request by csiyang**
# Description

Graphql already automatically refetches when query variables change. The `refetch` in code was making the query call twice, and causing issues to not properly load sometimes. This PR removes the superfluous refetch. 

- Link to Basecamp Todo

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] All visitor filters still work at `/journeys/[journeyId]/reports/visitors`

# Walkthrough

### <samp>🤖 Generated by Copilot at 5d89d45</samp>

*  Refetch visitors data whenever filter or sort settings change, using `fetchMore` function with updated variables ([link](https://github.com/JesusFilm/core/pull/1697/files?diff=unified&w=0#diff-e9e60ec48f7928f205dea4cd73b81cd0498ead851b3bf03af91d9a24b3055ba8L109-R109), [link](https://github.com/JesusFilm/core/pull/1697/files?diff=unified&w=0#diff-e9e60ec48f7928f205dea4cd73b81cd0498ead851b3bf03af91d9a24b3055ba8L149-R149), [link](https://github.com/JesusFilm/core/pull/1697/files?diff=unified&w=0#diff-e9e60ec48f7928f205dea4cd73b81cd0498ead851b3bf03af91d9a24b3055ba8R173))
*  Remove unused `useEffect` hook from imports in `visitors.tsx` ([link](https://github.com/JesusFilm/core/pull/1697/files?diff=unified&w=0#diff-e9e60ec48f7928f205dea4cd73b81cd0498ead851b3bf03af91d9a24b3055ba8L8-R8))
*  Add `visitorEdges` state to fetching condition to avoid null `after` variable in `visitors.tsx` ([link](https://github.com/JesusFilm/core/pull/1697/files?diff=unified&w=0#diff-e9e60ec48f7928f205dea4cd73b81cd0498ead851b3bf03af91d9a24b3055ba8L133-R133))

### CODE REVIEW

File: apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx [hunk 1]
```diff
@@ -208,6 +170,7 @@ function JourneyVisitorsPage(): ReactElement {
         setSortSetting('duration')
         break
     }
+    // await handleRefetchOnChange()
```
Comment (GeronimoJohn) [inline apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx:173]: remove?

File: apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx [hunk 2]
```diff
@@ -146,45 +146,7 @@ function JourneyVisitorsPage(): ReactElement {
     }
   }
 
-  useEffect(() => {
-    const handleRefetchOnChange = async (): Promise<void> => {
```
Comment (GeronimoJohn) [inline apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx:150]: the tests didn't get affected?

File: apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx (hunk 1 above)
Comment (csiyang) [reply apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx:173]: oh yep good catch

File: apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx (hunk 2 above)
Comment (csiyang) [reply apps/journeys-admin/pages/journeys/[journeyId]/reports/visitors.tsx:150]: nah, it actually is still being called, this useEffect was actually being double called before

**Review by GeronimoJohn — APPROVED** [review]

**Pull request merged**